dvc/ignore.py

```diff
@@ -131,20 +131,19 @@ def __init__(self, tree, root_dir):
         self.ignores_trie_tree[root_dir] = DvcIgnorePatterns(
             default_ignore_patterns, root_dir
         )
-        for root, dirs, _ in self.tree.walk(
-            self.root_dir, use_dvcignore=False
-        ):
-            self._update(root)
-            self._update_sub_repo(root, dirs)
-            dirs[:], _ = self(root, dirs, [])
+        self._update(self.root_dir)
 
     def _update(self, dirname):
+        old_pattern = self.ignores_trie_tree.longest_prefix(dirname).value
```

Comment:
Maybe here we also need 
```
        ignore_pattern = self.ignores_trie_tree.get(dirname)
        if ignore_pattern:
            return
```
to prevent it from running multiply times, when we are calling `_get_trie_pattern` twice with one dir and its child dir.

Reply:
Good point! Indeed, for now `_update` is only called in `_get_trie_pattern`, so that is already handled for us.